### PR TITLE
codeintel: Break out WAU counts in pings

### PR DESCRIPTION
**Pull request by efritz**
We send back the number of weekly active users that have had _any_ code intel event (precise or search-based). This sends additional counts for weekly active users getting precise events, search-based events, precise xrepo events, and search-based xrepo events.

Fixes https://github.com/sourcegraph/sourcegraph/issues/17923. Sister commit that updates the BigQuery schema in https://github.com/sourcegraph/analytics/pull/96.

### CODE REVIEW

File: internal/types/codeintel.go [hunk 1]
```diff
@@ -18,9 +18,14 @@ type CodeIntelAggregatedEvent struct {
...
-	StartOfWeek    time.Time
-	WAUs           int32
-	EventSummaries []CodeIntelEventSummary
+	StartOfWeek                    time.Time
+	WAUs                           *int32
+	PreciseWAUs                    *int32
+	SearchBasedWAUs                *int32
+	CrossRepositoryWAUs            *int32
+	PreciseCrossRepositoryWAUs     *int32
+	SearchBasedCrossRepositoryWAUs *int32
```
Comment (Strum355) [inline internal/types/codeintel.go:27]: Does this one make sense to have? Wasnt aware search-based had xrepo :thinking:

Comment (efritz) [reply internal/types/codeintel.go:27]: You can still get results from other repos via search. The reason we track xrepo metrics at all was for a prospect and sure admin wow factor. Maybe @ebrodymoore can comment here.

**Review by macraig — APPROVED** [review]
Can't wait to have WAUs back on the dashboard! 🤩

**Review by ebrodymoore — APPROVED** [review]
This is ready to be merged. The prod db has been updated!

**Comment by efritz** [comment]
Bonk.

**Pull request merged**